java/ddlogapi/DDlogConfig.java

```diff
@@ -0,0 +1,214 @@
+package ddlogapi;
+
+/**
+ * Configuration of a DDlog program instance.
+ */
+public class DDlogConfig {
+    /**
+     * Create default configuration.
+     */
+    public DDlogConfig() {
+        this.numTimelyWorkers = 1;
+        this.enableDebugRegions = false;
+        this.differentialIdleMergeEffort = 0;
+        this.profilingConfig = profilingDisabled();
+    }
+
+    public DDlogConfig(int workers) {
+        this();
+        this.numTimelyWorkers = workers;
+    }
+
+    public DDlogConfig(
+            int workers,
+            boolean enableDebugRegions,
+            long differentialIdleMergeEffort,
+            ProfilingConfig profilingConfig) {
+        this.numTimelyWorkers = workers;
+        this.enableDebugRegions = enableDebugRegions;
+        this.differentialIdleMergeEffort = differentialIdleMergeEffort;
+        this.profilingConfig = profilingConfig;
+    }
+
+    /**
+     * Logging mode for timely and differential profiling event streams.
+     * WARNING: The following constants are hardcoded in ddlogapi.c
+     */
+    public enum LogMode {
```

Comment:
enum again - i know you're patching that here, but they should really be that way at the source (?)